src/main/java/slimeknights/tconstruct/library/client/RenderUtil.java

```diff
@@ -173,7 +175,16 @@ public static void renderStackedFluidCuboid(FluidStack fluid, double px, double
     }
 
     int xd = to.getX() - from.getX();
-    int yd = (int) (ymax - ymin);
+
+    // the liquid can stretch over more blocks than the subtracted height is if ymin's decimal is bigger than ymax's decimal (causing UV over 1)
```

Comment:
I will get a screenshot as soon as I can, but basicit the if on the top quad gets messed up.

If you see my comment below, before this fix with min of 2.3 and max of 5.1 would get a `yd` of 2 (floored from 2.8, set is {2.3, 3, 4, 5.1}).

After the fix, you get a `yd` of 3 (floored from 3.1, set is {2.3, 3, 4, 5, 5.1})